src/EditorFeatures/Core/Shared/Utilities/IWorkspaceContextService.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.Editor.Shared.Utilities
+{
+    internal interface IWorkspaceContextService : IWorkspaceService
+    {
+        /// <summary>
+        /// Used to determine if running as a client in a cloud connected environment.
+        /// </summary>
+        bool IsInRemoteClientContext();
```

Comment:
❗ This is a really bad name. Based on this name, I came into this review to see how a service was implemented that distinguishes between in-proc and out-of-proc, only to find that it's completely unrelated.